plugin/reload/reload.go

```diff
@@ -0,0 +1,66 @@
+package reload
+
+import (
+	"crypto/md5"
+	"fmt"
+	"time"
+
+	"github.com/mholt/caddy"
+)
+
+// reload periodically checks if the Corefile has changed, and reloads if so
+
+type reload struct {
+	instance   *caddy.Instance
+	interval   time.Duration
+	sum        [md5.Size]byte
+	serverType string
```

Comment:
why do we need serverType? which for us is always 'dns'?

Reply:
oops. I ended up getting it from Caddyfile Input. The way this is written it could be used in any server type.